client/src/pages/ProductDetails.tsx

```diff
@@ -1,5 +1,109 @@
-import { Link } from 'react-router-dom';
+import { toDollars } from '../library/to-dollars';
+import { useParams } from 'react-router-dom';
+import type { Product } from '../library/data';
+import { useEffect, useState, useContext } from 'react';
+import { closeContext } from '../components/NavDrawerCloseContext';
 
 export function ProductDetails() {
-  return <Link to={'/'}>wya?</Link>;
+  const { productId } = useParams();
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [product, setProduct] = useState<Product>();
+  const { closeNavDrawer } = useContext(closeContext);
+
+  useEffect(() => {
+    async function loadProduct(productId: number) {
+      try {
+        const response = await fetch(`/api/catalog/details/${productId}`);
+        if (!response.ok) throw new Error(`fetch Error ${response.status}`);
+        const result = await response.json();
+        setProduct(result);
+      } catch (error) {
+        setError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    loadProduct(Number(productId));
+  }, [productId]);
+
+  if (isLoading) return <div>Loading...</div>;
+  if (error) {
+    return (
+      <div>
+        Error Loading catalog:{' '}
+        {error instanceof Error ? error.message : 'Unknown Error'}
+      </div>
+    );
+  }
+
+  if (!product) return null;
+  const { brand, name, amount, image, details } = product;
+  return (
+    <div onClick={closeNavDrawer} className="container">
+      <div className="flex  text-primary bg-secondary rounded shadow-wrapper">
+        <div className="ml-3">
+          <img
+            className="object-contain mt-3  object-contain"
+            src={image}
+            alt={name}
+          />
+          <h1 className="text-3xl font-bold my-3">{`${brand} ${name}`}</h1>
+          <h3>{toDollars(amount)}</h3>
+        </div>
+        <div className="mx-6">
+          <h1 className="text-3xl font-bold">Details</h1>
+          <p className="whitespace-normal">{details}.</p>
```

Comment:
The ending period is weird here. Shouldn't that be in the database instead?

It looks like it is there in some of the database entries. Those ones will end with a double period.